pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -28,6 +29,9 @@
 
   metalSupport ? stdenv.isDarwin && stdenv.isAarch64,
 
+  # You can find a full list here: https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
+  nvidiaArch ? "",
```

Comment:
Hi! Please choose the defaults based on `config.cudaSupport` and `config.cudaCapabilities`

Reply:
Similarly, the `march`/`mtune` could be configured using `stdenv.hostPlatform`. It might not even be necessary to pass this to the build script because the cc-wrapper probably already reflects `hostPlatform`

Reply:
> Similarly, the march/mtune could be configured using stdenv.hostPlatform 

Good catch, I added logic to determine which march to use at eval time.

I also did a bit of research and turns out `mtune` is not used on Apple Silicon but rather it `-mcpu=native` is used instead.

Reply:
> Hi! Please choose the defaults based on `config.cudaSupport` and `config.cudaCapabilities` 

Does Nix know about the host `cudaCapabilities` (As in which one the host is currently using)? 

If yes, it would make it possible to remove this as an option and instead dynamically assign it at `makeFlags`

Reply:
> Does Nix know about the host cudaCapabilities (As in which one the host is currently using)?

Nix itself doesn't know CUDA exists, but the user is expected to configure Nixpkgs at import time, like `import <nixpkgs. { config.cudaCapabilities = [ .. .]; }`. By default `cudaCapabilities` will be set to the list of all architectures supported by a particular cudatoolkit release for the platform

Reply:
> If yes, it would make it possible to remove this as an option and instead dynamically assign it at makeFlags

There actually is a plan to extend `cudaPackages.setupCudaHook` to set the relevant flags automatically (like `CMAKE_CUDA_ARCHITECTURES`, etc), but right now downstream derivations are expected to set these ad hoc

Reply:
> There actually is a plan to extend `cudaPackages.setupCudaHook` to set the relevant flags automatically (like `CMAKE_CUDA_ARCHITECTURES`, etc), but right now downstream derivations are expected to set these ad hoc

I guess until that becomes a thing, I could use a hack to expect an array with one string in it, match it with `lib.any` against `cudaPackages.cudaFlags.cudaCapabilities`, and pass it down. I think this would still comply with the `config.cudaCapabilities = [..];` part.

Do you have any suggestions, or is that good enough for now?

Reply:
> I guess until that becomes a thing, I could use a hack to expect an array with one string in it 

Yeah that's not what usually happens; what we do in other derivations that enforce using a single gencode is we take `buitlins.head cudaCapabilites` and hope that includes PTX